pir/cpp/client_test.cpp

```diff
@@ -263,78 +271,113 @@ TEST_F(PIRClientTest, TestProcessResponse) {
+  vector<Ciphertext> ct(1);
+  Encryptor()->encrypt(pt, ct[0]);
+
+  Response response;
+  SaveCiphertexts(ct, response.add_reply());
+  SaveCiphertexts(ct, response.add_reply());
+
+  auto result = client_->ProcessResponse(response).ValueOrDie();
+  EXPECT_THAT(result, ElementsAre(value, value));
 }
 
 TEST_F(PIRClientTest, TestCreateRequest_InvalidIndex) {
-  auto request_or = client_->CreateRequest(db_size_ + 1);
+  auto request_or = client_->CreateRequest({db_size_ + 1});
   ASSERT_EQ(request_or.status().code(),
             private_join_and_compute::StatusCode::kInvalidArgument);
 }
 
-class CreateRequestTest
-    : public PIRClientTest,
-      public testing::WithParamInterface<tuple<size_t, size_t, uint64_t>> {};
+class CreateRequestTest : public PIRClientTest,
+                          public testing::WithParamInterface<
+                              tuple<size_t, vector<size_t>, uint64_t>> {};
 
-TEST_P(CreateRequestTest, TestCreateRequest_MoreThanOneCT) {
+TEST_P(CreateRequestTest, TestCreateRequest) {
   const auto dbsize = get<0>(GetParam());
-  int desired_index = get<1>(GetParam());
+  vector<size_t> indexes = get<1>(GetParam());
   SetUpDB(dbsize);
 
   const auto poly_modulus_degree =
       pir_params_->GetEncryptionParams().poly_modulus_degree();
   const auto plain_mod =
       pir_params_->GetEncryptionParams().plain_modulus().value();
 
-  auto request_or = client_->CreateRequest(desired_index);
+  auto request_or = client_->CreateRequest(indexes);
   ASSERT_TRUE(request_or.ok())
       << "Status is: " << request_or.status().ToString();
-  auto query =
-      LoadCiphertexts(Context()->SEALContext(), request_or.ValueOrDie().query())
-          .ValueOrDie();
-  ASSERT_EQ(query.size(), dbsize / poly_modulus_degree + 1);
-  EXPECT_THAT(request_or.ValueOrDie().galois_keys(), Not(IsEmpty()));
-
-  for (const auto& ct : query) {
-    Plaintext pt;
-    Decryptor()->decrypt(ct, pt);
-    for (size_t i = 0; i < pt.coeff_count(); ++i) {
-      if (i != static_cast<size_t>(desired_index)) {
-        EXPECT_EQ(pt[i], 0);
-      }
-    }
-    if (desired_index < 0 ||
-        static_cast<size_t>(desired_index) >= poly_modulus_degree) {
-      desired_index -= poly_modulus_degree;
+
+  auto request = request_or.ValueOrDie();
+  ASSERT_EQ(request.query_size(), indexes.size());
+  EXPECT_THAT(request.galois_keys(), Not(IsEmpty()));
+
+  auto m = get<2>(GetParam());
+
+  for (size_t idx = 0; idx < indexes.size(); ++idx) {
+    auto query = LoadCiphertexts(Context()->SEALContext(), request.query(idx))
+                     .ValueOrDie();
+    ASSERT_EQ(query.size(), dbsize / poly_modulus_degree + 1);
+    int desired_index = indexes[idx];
+
+    for (const auto& ct : query) {
+      Plaintext pt;
+      Decryptor()->decrypt(ct, pt);
       for (size_t i = 0; i < pt.coeff_count(); ++i) {
-        EXPECT_EQ(pt[i], 0);
+        if (i != static_cast<size_t>(desired_index)) {
+          EXPECT_EQ(pt[i], 0);
```

Comment:
I'm not sure what I was originally thinking here, but this for loop is in the wrong place. It should probably be under the else clause below.